main/plots.py

```diff
@@ -111,229 +117,73 @@ def create_capacity_planning_plot(start_date: datetime, end_date: datetime) -> R
         end_date: datetime object representing the end of the plotting period
         x_range: (optional) tuple of datetimes to use as the x_range for the displayed
             plot
+        selected_projects: (optional) list of selected project name(s)
+        selected_users: (optional) list of selected user name(s)
 
     Returns:
-        Bokeh Row layout with timeseries data plot and CheckboxGroup widgets.
+        Bokeh figure layout with timeseries data plot and MultiChoice widgets.
 
     """
-    # Get effort for each project and user
-    PROJECTS = ["proCAT 1", "proCAT 2", "proCAT 3", "proCAT 4"]
-    USERS = ["User 1", "User 2", "User 3", "User 4", "User 5"]
-    project_colours = ["firebrick", "orange", "green", "blue"]
-    user_colours = ["purple", "brown", "pink", "cyan", "magenta"]
+    # collect project names and users'm names
+    all_projects = list(
+        models.Project.objects.values_list("name", flat=True).distinct()
+    )
+    all_users = list(models.User.objects.values_list("username", flat=True).distinct())
+
+    if not selected_projects:
+        use_projects = all_projects
+    else:
+        use_projects = selected_projects
 
-    # Total effort and capacity across all projects
-    total_effort_timeseries = timeseries.get_effort_timeseries(start_date, end_date)
-    capacity_timeseries = timeseries.get_capacity_timeseries(start_date, end_date)
+    if not selected_users:
+        use_users = all_users
+    else:
+        use_users = selected_users
```

Comment:
Then this can be simplified a bit (it does not let me add a suggestion; sorry):
```python
if not use_projects:
    use_projects = list(
        models.Project.objects.values_list("name", flat=True).distinct()
    )
if not use_users:
    use_users = list(models.User.objects.values_list("username", flat=True).distinct())
```